src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1268,80 +1290,121 @@ static bool IsRedirectToGet(HttpStatusCode code)
+                    (response.Content.Headers.ContentRange.HasLength &&
+                    response.Content.Headers.ContentRange.Length != _resumeFileSize))
                 {
-                    response = GetResponse(client, redirectRequest, keepAuthorization);
-                }
-            }
+                    _cancelToken.Cancel();
 
-            // Request again without the Range header because the server indicated the range was not satisfiable.
-            // This happens when the local file is larger than the remote file.
-            // If the size of the remote file is the same as the local file, there is nothing to resume.
-            if (Resume.IsPresent &&
-                response.StatusCode == HttpStatusCode.RequestedRangeNotSatisfiable &&
-                (response.Content.Headers.ContentRange.HasLength &&
-                response.Content.Headers.ContentRange.Length != _resumeFileSize))
-            {
-                _cancelToken.Cancel();
+                    WriteVerbose(WebCmdletStrings.WebMethodResumeFailedVerboseMsg);
+
+                    // Disable the Resume switch so the subsequent calls to GetResponse() and FillRequestStream()
+                    // are treated as a standard -OutFile request. This also disables appending local file.
+                    Resume = new SwitchParameter(false);
 
-                WriteVerbose(WebCmdletStrings.WebMethodResumeFailedVerboseMsg);
+                    using (HttpRequestMessage requestWithoutRange = GetRequest(currentUri))
+                    {
+                        FillRequestStream(requestWithoutRange);
+                        long requestContentLength = 0;
+                        if (requestWithoutRange.Content != null)
+                        {
+                            requestContentLength = requestWithoutRange.Content.Headers.ContentLength.Value;
+                        }
 
-                // Disable the Resume switch so the subsequent calls to GetResponse() and FillRequestStream()
-                // are treated as a standard -OutFile request. This also disables appending local file.
-                Resume = new SwitchParameter(false);
+                        string reqVerboseMsg = String.Format(
+                            CultureInfo.CurrentCulture,
+                            WebCmdletStrings.WebMethodInvocationVerboseMsg,
+                            requestWithoutRange.Method,
+                            requestWithoutRange.RequestUri,
+                            requestContentLength);
+                        WriteVerbose(reqVerboseMsg);
 
-                using (HttpRequestMessage requestWithoutRange = GetRequest(currentUri))
+                        return GetResponse(client, requestWithoutRange, keepAuthorization);
+                    }
+                }
+
+                _resumeSuccess = response.StatusCode == HttpStatusCode.PartialContent;
+
+                if (ShouldRetry(response.StatusCode))
```

Comment:
The logic here results in the verbose message printed an extra time when the final result is still an error:

```powershell
Invoke-RestMethod -RetryCount 1 -RetryIntervalSec 1 https://httpbin.org/status/404 -Verbose -SkipCertificateCheck
```

```none
VERBOSE: GET https://httpbin.org/status/404 with 0-byte payload
VERBOSE: Retrying after interval of 1 seconds. Status code for previous attempt: NotFound
VERBOSE: Retrying after interval of 1 seconds. Status code for previous attempt: NotFound
VERBOSE: received 0-byte response of content type text/html
irm : Response status code does not indicate success: 404 (NOT FOUND).
At line:1 char:1
+ irm -RetryCount 1 -RetryIntervalSec 1 https://httpbin.org/status/404  ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidOperation: (Method: GET, Re...rShell/6.1.0
}:HttpRequestMessage) [Invoke-RestMethod], HttpResponseException
+ FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
```

Maybe `if (totalRequests<0 && ShouldRetry(response.StatusCode))`?

Reply:
Fixed.